src/app/blocks/map-block/map-block.component.ts

```diff
@@ -28,23 +29,92 @@ contributors, <a href="https://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA
   markerClusterData = [];
   markerClusterOptions = {};
 
+  // Country GeoJSON provider base URL
+  countryProviderBaseURL = 'https://raw.githubusercontent.com/AshKyd/geojson-regions/main/public/countries/50m/';
+  countryExtension = '.geojson';
+
+  // Country area polygon styling
+  countryStyle = {
+    color: 'gray',
+    weight: 2,
+    opacity: 0.3,
+    fillColor: 'gray',
+    fillOpacity: 0.5
+  };
+
+  // In-memory cache for country GeoJSON data
+  countryCache: { [key: string]: any } = {};
```

Comment:
Do we need to use an `any` here?

Reply:
The object is being mutated, so I instinctively typed it as 'any' to avoid any issues. A different format could be possible.